### PR TITLE
[crd-generator] cycle in base java classes on Java 19+

### LINKED ISSUE #5866: CRD generation fails when compiled using java version 19 and above
### Describe the bug

CRD generation fails when compiled using java version 19 and above.

### Fabric8 Kubernetes Client version

SNAPSHOT

### Steps to reproduce

There is an example project here, CRD generation succeeds when built with java version 18 and fails when built with java version 19.

https://github.com/matteriben/crd-generator-date-time-reproducer/actions/runs/8583665846

[job-logs.txt](https://github.com/fabric8io/kubernetes-client/files/14894938/job-logs.txt)

### Expected behavior

CRD generation is expected to succeed when project is built with java version 19 and above.

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

1.25.3@latest

### Environment

Linux

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

This seems to be related to a change in the implementation of java.time.ZonedDateTime from version 18 to 19 and above.

### PR DESCRIPTION
**Pull request by andreaTP**
## Description

Proposal to attempt to fix #5866 

## Type of change
 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by andreaTP** [comment]
A possible next step would be to define a List of `DEFAULT_SCHEMA_SWAPS` to be applied always ... but I want feedback on this idea as I'm not 100% convinced myself.

**Comment by shawkins** [comment]
> A possible next step would be to define a List of `DEFAULT_SCHEMA_SWAPS` to be applied always ... but I want feedback on this idea as I'm not 100% convinced myself.

Please don't go down this path. It still produces a schema that is not correct and adds new internal usage of schema swap that we don't need - you could just as easily handle this with a common mapping at the level of ZonedDateTime. But ultimately you'll end up doing that for a great number of classes - this is very brittle.

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/InternalSchemaSwaps.java [hunk 1]
```diff
@@ -94,8 +102,14 @@ public SwapResult lookupAndMark(ClassRef originalType, String name) {
     return new SwapResult(null, false);
   }
 
+  private static boolean isIgnored(Value value) {
```
[the remark this answers is not in the document]
Comment (shawkins) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/InternalSchemaSwaps.java:105]: I'm not sure what you mean here at first glance - is there a test case that could highlight what you are describing?

[the remark this answers is not in the document]
Comment (andreaTP) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/InternalSchemaSwaps.java:105]: If you remove the logic of this method and `return false` you have a reproducer of the issue.

**Comment by andreaTP** [comment]
> Please don't go down this path. It still produces a schema that is not correct and adds new internal usage of schema swap that we don't need - you could just as easily handle this with a common mapping at the level of ZonedDateTime. But ultimately you'll end up doing that for a great number of classes - this is very brittle.

Fair assessment, should we keep it as "manual" workaround?
Do we have other alternatives?

Open to ideas!

**Comment by shawkins** [comment]
> Fair assessment, should we keep it as "manual" workaround?

The manual workaround is to a swap on the entire ZonedDateTime.

> Do we have other alternatives?

As mentioned in the other PR we can either use Jackson's schema logic - which will work automagically most of the time and keep redundant annotations to a minimum. Or we can error out if we don't think we'll produce the approriate schema. There are two paths to that - either we still use Jackson's schema logic and detect when it has something other than an object, or start blessing or marking packages / classes as supported - most of the problems are going to be with java / javax. That latter one feels too britle as well.

**Comment by andreaTP** [comment]
> The manual workaround is to a swap on the entire ZonedDateTime.

I'm ok with this workaround, the alternatives seem to bring additional complexity, but happy to hear more opinions/feedback/proposals.

**Comment by andreaTP** [comment]
Superseded by #5877

**Pull request closed without merging**